### PR TITLE
Events-only endpoints for address and tx_id

### LINKED ISSUE #875: Expose events-only tx endpoint
Hi there!

Currently when i'm working with transaction data, (eg fetching all transactions for a given principal), none of the events data is included in the transaction object. As such I need to do a secondary fetch for each transaction to the `/extended/v1/tx` endpoint to grab only the events.

I was thinking it'd be really helpful to have an endpoint at something like `/extended/v1/tx-events` that could take either a single `tx_id`, or an array of `tx_ids`.

Curious what you all think, or if there is a better solution!

### PR DESCRIPTION
**Pull request by asimm241**
## Description
Add endpoints to return events for principal or transaction-id 

closes #875 #883 

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
_NO_

## Testing information

API unit tests are added

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `npm run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @rafaelcr

### CODE REVIEW

File: docs/openapi.yaml
Comment (rafaelcr) [inline docs/openapi.yaml:3173]: These two endpoints should be mixed into one general-purpose endpoint that allows you to filter for events by principal, tx_id, type, etc:
```
/extended/v1/tx/events
```

File: docs/openapi.yaml
Comment (rafaelcr) [inline docs/openapi.yaml:3178]: Move to `Transactions`

File: docs/openapi.yaml
Comment (rafaelcr) [inline docs/openapi.yaml:3184]: Add a `tx_id` parameter. Users should specify at least one of these two.

File: docs/openapi.yaml
Comment (rafaelcr) [inline docs/openapi.yaml:3215]: Remove this endpoint

File: src/api/routes/events.ts
Comment (rafaelcr) [inline src/api/routes/events.ts:35]: Move these functions to the Transactions router

File: src/datastore/postgres-store.ts
Comment (rafaelcr) [inline src/datastore/postgres-store.ts:4352]: A few comments on these queries:
1. Querying for events in separate `query()` calls is a bad idea for pagination because you want to paginate the results, not filter against the `event_index` property of every table. You should switch to a single query in which you make `UNION` of all the results and then paginate over those **sorted** results (it will be very important to sort by `block_height, microblock_sequence, tx_index, event_index`).
2. Make sure to filter both `canonical` AND `microblock_canonical`

**Review by rafaelcr — CHANGES_REQUESTED** [review]

File: src/datastore/postgres-store.ts
Comment (asimm241) [reply src/datastore/postgres-store.ts:4352]: I followed the same approach as this https://github.com/hirosystems/stacks-blockchain-api/blob/f1abc291d955525983d40da11f8f5c638cee756c/src/datastore/postgres-store.ts#L4077

should we look into this as well?

Comment (rafaelcr) [reply src/datastore/postgres-store.ts:4352]: Interesting @asimm241.

Where is that function being used? We could evaluate whether or not to refactor that one (later, in another PR) based on the current use.

Comment (asimm241) [reply src/datastore/postgres-store.ts:4410]: @rafaelcr I don't think we can do union here since all events have different columns.

File: src/api/init.ts
Comment (zone117x) [inline src/api/init.ts:13]: Can you revert unnecessary whitespace / stylistic changes like this?

File: src/api/query-helpers.ts
Comment (zone117x) [inline src/api/query-helpers.ts:233]: This should be `req.query.tx_id` rather than `req.query.txId` right?

File: src/api/query-helpers.ts
Comment (zone117x) [inline src/api/query-helpers.ts:273]: The try/catch around this whole block is meant to handle the error thrown by `parseEventTypeStrings()` right? If so, it should be scoped to that function call such that it doesn't overlap with the other `handleBadRequest` instead of the try body.

File: src/api/routes/tx.ts
Comment (zone117x) [inline src/api/routes/tx.ts:263]: Seems like `address` and `txId` input should be validated here for 400 (bad request) before handing off to datastore layer.

File: docs/openapi.yaml
Comment (rafaelcr) [inline docs/openapi.yaml:3180]: Add an example to each of these parameters

**Review by rafaelcr — CHANGES_REQUESTED** [review]

File: docs/api/transaction/get-transaction-events.example.json
Comment (rafaelcr) [inline docs/api/transaction/get-transaction-events.example.json:4]: Please change this example to be a real response instead of using mock data ("0x1234", etc.)

File: docs/generated.d.ts
Comment (rafaelcr) [inline docs/generated.d.ts:766]: Can you remove the a 1 at the end here? I think you'll need to edit the TransactionEvent entity schema somewhere

File: docs/api/transaction/get-transaction-events.example.json
Comment (rafaelcr) [reply docs/api/transaction/get-transaction-events.example.json:4]: Is this a real response? If so, why are the `event_index` entries out of order for the same tx?

File: docs/openapi.yaml
Comment (rafaelcr) [inline docs/openapi.yaml:3188]: Add the `0x` at the start for consistency

File: src/datastore/common.ts
Comment (rafaelcr) [inline src/datastore/common.ts:683]: Can you add a JSdoc comment to this function?
```js
/**
 * Example...
 * @param ...
 */
```

**Review by rafaelcr — CHANGES_REQUESTED** [review]
It's looking very good, just a few style comments

File: src/datastore/postgres-store.ts
Comment (rafaelcr) [inline src/datastore/postgres-store.ts:4383]: Looks good, but move it to `common.ts` so the comment appears in tooltips everywhere

File: src/api/routes/tx.ts
Comment (rafaelcr) [inline src/api/routes/tx.ts:273]: Can you add the chain tip cache headers to this response? You can look at the `/block` endpoint for an example on how to do that

File: src/tests/api-tests.ts
Comment (rafaelcr) [inline src/tests/api-tests.ts:9815]: Set the test name to "/tx/events" so we can find it easily

File: src/tests/api-tests.ts
Comment (rafaelcr) [inline src/tests/api-tests.ts:9898]: "/tx/events empty events returned" and so on

**Review by rafaelcr — CHANGES_REQUESTED** [review]
Almost there

**Review by rafaelcr — APPROVED** [review]
LGTM 🏁 
Can you hold off on merging until we've shipped this next release though?

**Comment by asimm241** [comment]
@rafaelcr bumping it up! this PR is rebased and ready to merge.

**Review by rafaelcr — CHANGES_REQUESTED** [review]
One small style nit before merging

**Review by rafaelcr — APPROVED** [review]
Great @asimm241 thank you

**Comment by rafaelcr** [comment]
Can you fix the new conflicts so we can merge, please? @asimm241

**Pull request merged**
